tests/e2e/query_planning/query_planning_cartesian.py

```diff
@@ -38,5 +38,37 @@ def test_indexed_join_with_indices(memgraph):
     assert expected_explain == actual_explain
 
 
+def test_indexed_join_with_indices_split(memgraph):
+    memgraph.execute("CREATE INDEX ON :Label1(Prop1);")
+
+    expected_explain = [
+        " * Produce {a0, n0, n1, n2, n3, n4, n5, r0, r1, r2}",
+        " * Expand (n5)-[r2]->(n4)",
+        " * Filter (n5 :Label0:Label1)",
+        " * ScanAll (n5)",
+        " * Unwind",
+        " * EdgeUniquenessFilter {r0 : r1}",
+        " * IndexedJoin",
+        " |\\ ",
+        " | * Expand (n0)<-[r0]-(n1)",
+        " | * ScanAllByLabelPropertyRange (n0 :Label1 {Prop1})",
+        " | * Once",
+        " * Filter (n3 :Label0)",
+        " * Expand (n2)<-[r1]-(n3)",
+        " * Filter (n2 :Label1)",
+        " * ScanAll (n2)",
+        " * Once",
+    ]
+
+    results = list(
+        memgraph.execute_and_fetch(
+            "EXPLAIN MATCH (n0 :Label1)<-[r0]-(n1), (n2 :Label1)<-[r1]-(n3 :Label0) UNWIND [1] AS a0 MATCH (n4)<-[r2]-(n5 :Label0 :Label1) WHERE (((n2.Prop0) > (n0.Prop1)))  RETURN *"
+        )
```

Comment:
Where is this n2.Prop0 > n0.Prop1 evaluated, I can't seem to see it from the explain :eyes:

Reply:
It is inside the `ScanAllByLabelPropertyRange` filter. The printout isn't verbose...

Reply:
Why does the query need to be this complex?

Reply:
This is the query that broke our planner/optimizer.
It has to be so complex because otherwise our planner works :smile: 
What happened is that the `Unwind` got between the `Filter {n2.Prop0 > n0.Prop1}` being embedded and the `EdgeUniquenessFilter`.
Before if the `EdgeUniquenessFilter` was right before the `Filter`, we would know that a `Cartesian` was being used and would swap it with a `Join`. 
This logic broke once the `Unwind` was introduced.
This PR tries to make sure all queries that need to swap to a Join do so.